src/compas_model/model/group_node.py

```diff
@@ -48,3 +50,193 @@ def __init__(self, name=None, geometry=None, attributes=None, parent=None):
+    # Printing
+    # ==========================================================================
+
+    def __repr__(self):
+        return "<{}> {}, <geometry> {}".format(
+            self.__class__.__name__, self.name, self.geometry
+        )
+
+    def __str__(self):
+        return self.__repr__()
+
+    def add_element(
+        self, name=None, element=None, attributes=None, copy_element=False, parent=None
+    ):
+
+        """Add element to the group
+
+        Triple Behavior:
+
+        1. If Element_Node exists, add an element to the base dictionary of the Model class.
+        2. If Element_Node exists, add an element to the graph.
+        3. Add a node tree.
+
+        Parameters
+        ----------
+        name : str
+            a name of the ElementNode.
+        element : Element
+            Element object or any class that inherits from Element class.
+        attributes : dict, optional
+            A dictionary of additional attributes to be associated with the node.
+        copy_element : bool, optional
+            If True, the element is copied before adding to the tree.
+        parent : Node, optional
+            The parent node of this node.
+
+        """
+        # -----------------------------------------------------------------------
+        # user interface
+        # 1 user did not provide anything -> return None
+        # 2 user provided element input -> my_node.add_element(name, element)
+        # 3 otherwise there is an exception
+        # -----------------------------------------------------------------------
+
+        if isinstance(element, Element):
+            element_copy = element.copy() if copy_element else element
+            name = name if name else str(element_copy.guid)
+        else:
+            raise Exception("At least provide element input")
+
+        # -----------------------------------------------------------------------
+        # set a parent property in element, incase you want to traverse backwards
+        # create a node and add it to the tree
+        # -----------------------------------------------------------------------
+        element_copy.parent = parent if parent else self
+        node = ElementNode(
+            name=name,
+            element=element_copy,
+            attributes=attributes,
+            parent=element_copy.parent,
+        )
+
+        # -----------------------------------------------------------------------
+        # add the node to the children list
+        # -----------------------------------------------------------------------
+        self._children.append(node)
+
+        # -----------------------------------------------------------------------
+        # add elements to the base dictionary of Model class and graph
+        # -----------------------------------------------------------------------
+        if self._tree is not None:
+            if self._tree._model is not None:
```

Comment:
i think you should avoid interacting with the protected attributes of attributes.
it might be that there is an entire mechanism attached to the public property